websocket_message_test.go

```diff
@@ -76,231 +76,435 @@ func TestWebsocketMessage_Alias(t *testing.T) {
+			name:   "binary message type with BinaryMessage function",
+			fn:     msg.BinaryMessage,
+			testFn: msg.BinaryMessage,
+			want:   success,
+		},
+		{
+			name:   "binary message type with NotBinaryMessage function",
+			fn:     msg.BinaryMessage,
+			testFn: msg.NotBinaryMessage,
+			want:   failure,
+		},
+		{
+			name:   "binary message type with TextMessage function",
+			fn:     msg.BinaryMessage,
+			testFn: msg.TextMessage,
+			want:   failure,
+		},
+		{
+			name:   "binary message type with NotTextMessage function",
+			fn:     msg.BinaryMessage,
+			testFn: msg.NotTextMessage,
+			want:   success,
+		},
+	}
 
-	msg.TextMessage()
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.testFn().chain.assert(t, tc.want)
+			msg.chain.clear()
+		})
+	}
 
-	msg.NotTextMessage()
-	msg.chain.assert(t, success)
-	msg.chain.clear()
+	cases2 := []struct {
+		name        string
+		typ         int
+		testTyp     int
+		wantCode    chainResult
+		wantNotCode chainResult
+	}{
+		{
+			name:        "binary message type with BinaryMessage type",
+			typ:         websocket.BinaryMessage,
+			testTyp:     websocket.BinaryMessage,
+			wantCode:    success,
+			wantNotCode: failure,
+		},
+		{
+			name:        "binary message type with TextMessage type",
+			typ:         websocket.BinaryMessage,
+			testTyp:     websocket.TextMessage,
+			wantCode:    failure,
+			wantNotCode: success,
+		},
+	}
 
-	msg.Type(websocket.BinaryMessage)
-	msg.chain.assert(t, success)
-	msg.chain.clear()
+	for _, tc := range cases2 {
+		t.Run(tc.name, func(t *testing.T) {
+			reporter := newMockReporter(t)
 
-	msg.NotType(websocket.BinaryMessage)
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
+			NewWebsocketMessage(reporter, tc.typ, nil, 10).Type(tc.testTyp).
+				chain.assert(t, tc.wantCode)
 
-	msg.Type(websocket.TextMessage)
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
+			NewWebsocketMessage(reporter, tc.typ, nil, 10).NotType(tc.testTyp).
+				chain.assert(t, tc.wantNotCode)
+		})
 
-	msg.NotType(websocket.TextMessage)
-	msg.chain.assert(t, success)
-	msg.chain.clear()
+	}
 }
 
 func TestWebsocketMessage_MatchTypes(t *testing.T) {
-	reporter := newMockReporter(t)
-
-	msg := NewWebsocketMessage(reporter, websocket.TextMessage, nil, 0)
-
-	msg.Type(websocket.TextMessage, websocket.BinaryMessage)
-	msg.chain.assert(t, success)
-	msg.chain.clear()
-
-	msg.Type(websocket.BinaryMessage, websocket.TextMessage)
-	msg.chain.assert(t, success)
-	msg.chain.clear()
-
-	msg.Type(websocket.CloseMessage, websocket.BinaryMessage)
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
-
-	msg.Type(websocket.BinaryMessage, websocket.CloseMessage)
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
-
-	msg.NotType(websocket.TextMessage, websocket.BinaryMessage)
-	msg.chain.assert(t, failure)
-	msg.chain.clear()
+	cases := []struct {
+		name        string
+		typ         int
+		testTyp     int
+		testTyp2    int
```

Comment:
Instead of testTyp + testTyp2, I suggest to have a single slice testTypes and pass it to Type() using `...`.

Reply:
Same for tc1 and tc2 in TestWebsocketMessage_MatchCodes.

Reply:
done